processing/src/main/java/org/apache/carbondata/processing/loading/steps/DataWriterBatchProcessorStepImpl.java

```diff
@@ -85,8 +86,9 @@ public DataWriterBatchProcessorStepImpl(CarbonDataLoadConfiguration configuratio
           CarbonRowBatch next = iterator.next();
           // If no rows from merge sorter, then don't create a file in fact column handler
           if (next.hasNext()) {
+            DataMapWriterListener listener = getDataMapWriterListener(0);
             CarbonFactDataHandlerModel model = CarbonFactDataHandlerModel
-                .createCarbonFactDataHandlerModel(configuration, storeLocation, 0, k++);
+                .createCarbonFactDataHandlerModel(configuration, storeLocation, 0, k++, listener);
```

Comment:
Why  not create the listener inside the `createCarbonFactDataHandlerModel`?

Reply:
beacuse i needed listener in DataWriterProcessorStepImpl, to close all the DMwriters if the load is failed, so i created the listener outside and passed to function, which in turn will set this into model.